### PR TITLE
Update deps and prepare release of cargo-concordium and vscode ext

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

- Update dependencies to prepare for Concordium Protocol Version 7.
- Prepare releases for `cargo-concordium` version `4.0.0` and the vscode extension version `3.0.0`.
- Adjust VS Code extension, such that the new defaults of `cargo-concordium` does not break the already exposed commands.

### CODE REVIEW

File: cargo-concordium/CHANGELOG.md [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
+
+- Use smart contract v1 cost configuration when simulating locally (the reduce execution cost introduced as part of Concordium Protocol Version 7)
+- Added a `--tag <RELEASE-TAG>` option to `cargo concordium init` to allow specifying the release tag of the templates.
+  Defaults to `releases/templates/latest`.
 - When running integration tests the module output path is now exposed to the
   tests via the `CARGO_CONCORDIUM_TEST_MODULE_OUTPUT_PATH` environment variable.
-- Change the default build output path to `concordium-out/module.wasm.v1`.
-- Remove requirement for `--out` flag when using the `--verifiable` flag.
-- Embed the schema in the Wasm module by default. Therefore, the `--schema-embed` flag is now deprecated. This behavior can be disabled with the `--no-schema-embed` flag.
+- Introduce a default build output (`--out <PATH>`) path to `concordium-out/module.wasm.v1`.
```
Comment (DOBEN) [inline cargo-concordium/CHANGELOG.md:12]:
```suggestion
- Add the default build output (`--out <PATH>`) path `concordium-out/module.wasm.v1`.
```

File: cargo-concordium/CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,17 @@
...
+- Remove requirement for `--out` flag when using the `--verifiable` flag, now using the default output path.
```
Comment (DOBEN) [inline cargo-concordium/CHANGELOG.md:13]:
```suggestion
- Change `--out` flag being optional when using the `--verifiable` flag.
- Change `--verifiable` flag to use the default output path if the `--out` flag is not set.
```

**Review by DOBEN — APPROVED** [review]
LGTM 

Some small suggestions regarding the `changelog` only.

**Review by soerenbf — APPROVED** [review]
Looks good 😄

**Pull request merged**